src/sentry/flags/endpoints/logs.py

```diff
@@ -47,6 +57,26 @@ def serialize(self, obj, attrs, user, **kwargs) -> FlagAuditLogModelSerializerRe
         }
 
 
+class FlagLogIndexRequestSerializer(rest_serializers.Serializer):
+    # start, end handled separately.
+    flag = rest_serializers.ListField(
+        child=rest_serializers.CharField(),
+        required=False,
+    )
+    provider = rest_serializers.ListField(
+        child=rest_serializers.ChoiceField(choices=ProviderEnum.get_names() + ["unknown"]),
+        required=False,
+    )
+    sort = rest_serializers.CharField(required=False, allow_null=True)
+
+    def validate_provider(self, value: list[str]) -> list[int | None]:
+        return [(PROVIDER_MAP[provider] if provider != "unknown" else None) for provider in value]
+
+    # Support camel case since it's used by our response serializer.
+    def validate_sort(self, value: str | None) -> str | None:
+        return camel_to_snake_case(value) if value else None
```

Comment:
This would be a great place to check if the sort value is in a pre-approved list of fields.